src/Events/Custom_Tables/V1/WP_Query/Custom_Tables_Query.php

```diff
@@ -297,20 +301,49 @@ public function order_by_occurrence_id( $order_by, $query = null ) {
 			return $order_by;
 		}
 
-		remove_filter( 'posts_orderby', [ $this, 'order_by_occurrence_id' ] );
+		remove_filter( 'posts_orderby', [ $this, 'order_by_occurrence_id' ], 100 );
 
+		$original_order_by = $this->wp_query->query_vars['orderby'] ?? [];
+		$normalized_order_by = tribe_normalize_orderby( $original_order_by );
 		$occurrences = Occurrences::table_name( true );
-		global $wpdb;
 
-		/*
-		 * Replace, implicitly redirecting them, a curated list of order criteria to the Occurrence-table
-		 * based criteria.
-		 *
-		 * @todo is this code eligible for a more general purpose use? Should it be more flexible?
-		 */
+		if ( ! empty( $normalized_order_by ) ) {
+			global $wpdb;
+
+			// Rebuild the ORDER string based on the custom tables redirection.
+			$buffer = [];
+			$meta_query_clauses = $this->meta_query->get_clauses();
+			foreach ( $normalized_order_by as $original_key => $direction ) {
+				if ( $original_key === 'meta_value' ) {
+					// Handle queries with on meta value.
+					$original_key = array_key_first( $meta_query_clauses );
+				}
+
+				if ( in_array( $original_key, [ 'ID', $wpdb->posts . '.ID' ], true ) ) {
+					// If the order is by post ID, order by post ID and occurrence ID.
+					$buffer[] = "ID $direction, $occurrences.occurrence_id $direction";
```

Comment:
Shouldn't these be parameterized? `$wpdb->prepare($query, $table)` ?

Reply:
If we prepared the table name and the direction, then they would both be prepared as strings, it would result in a SQL string like `ID 'ASC', 'wp_tec_occurrences'.occurrence_id 'ASC'` that would be invalid SQL. SQL fragments like these are typically composed like this and in a controlled context. The table name comes from a function we control.